### PR TITLE
Fixes #100 Perform memory-leak analysis

### LINKED ISSUE #100: Perform memory-leak analysis
s. https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/1421

### PR DESCRIPTION
**Pull request by Yuri05**

### CODE REVIEW

File: src/OSPSuite.SimModel/Simulation.cs [hunk 1]
```diff
@@ -654,12 +654,9 @@ protected virtual void Dispose(bool disposing)
...
 
-         if (disposing)
-         {
-            SimulationImports.DisposeParameterInfoVector(_allParameters);
-            SimulationImports.DisposeSpeciesInfoVector(_allSpecies);
-            SimulationImports.DisposeSimulation(_simulation);
-         }
+         SimulationImports.DisposeParameterInfoVector(_allParameters);
+         SimulationImports.DisposeSpeciesInfoVector(_allSpecies);
+         SimulationImports.DisposeSimulation(_simulation);
```
Comment (Yuri05) [inline src/OSPSuite.SimModel/Simulation.cs]: That was wrong. _Unmanaged_ resources must be released **always**, not only `if disposing`.
s. also https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

File: src/OSPSuite.SimModelNative/OSPSuite.SimModelNative.vcxproj [hunk 2]
```diff
@@ -79,7 +79,7 @@
       <RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>
       <PrecompiledHeader>NotUsing</PrecompiledHeader>
       <WarningLevel>Level3</WarningLevel>
-      <DebugInformationFormat>OldStyle</DebugInformationFormat>
+      <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
```
Comment (Yuri05) [inline src/OSPSuite.SimModelNative/OSPSuite.SimModelNative.vcxproj]: Figured out that this setting was not required for the Visual Leak Detector. Only "whole program optimization" is relevant. Thus resetting to the previous value.

File: src/OSPSuite.SysTool/OSPSuite.SysTool.vcxproj [hunk 3]
```diff
@@ -66,7 +66,7 @@
       <RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>
       <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
-      <DebugInformationFormat>OldStyle</DebugInformationFormat>
+      <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
       <WholeProgramOptimization>true</WholeProgramOptimization>
```
Comment (Yuri05) [inline src/OSPSuite.SysTool/OSPSuite.SysTool.vcxproj]: same here.

File: src/OSPSuite.XMLWrapper/OSPSuite.XMLWrapper.vcxproj [hunk 4]
```diff
@@ -66,7 +66,7 @@
       <RuntimeLibrary>MultiThreadedDebugDLL</RuntimeLibrary>
       <PrecompiledHeader />
       <WarningLevel>Level3</WarningLevel>
-      <DebugInformationFormat>OldStyle</DebugInformationFormat>
+      <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
       <WholeProgramOptimization>true</WholeProgramOptimization>
```
Comment (Yuri05) [inline src/OSPSuite.XMLWrapper/OSPSuite.XMLWrapper.vcxproj]: and here

File: src/OSPSuite.SimModel/Simulation.cs (hunk 1 above)
Comment (msevestre) [reply src/OSPSuite.SimModel/Simulation.cs]: indeed. I don't think i ever used this variable in code, Interesting

**Pull request merged**